realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1415,8 +1421,9 @@ private void checkMaxDepth(int maxDepth) {
      * version, nothing will happen.
      *
      * @param configuration {@link RealmConfiguration}
+     * @throws FileNotFoundException if the Realm file doesn't exist.
      */
-    public static void migrateRealm(RealmConfiguration configuration) {
+    public static void migrateRealm(RealmConfiguration configuration) throws FileNotFoundException {
```

Comment:
Yeah, I got to the same conclusion after found `FileNotFoundException` is a checked exception. It does make sense to be checked.